### PR TITLE
Refactor code that looks at the videos directory and gets next video 

### PR DESCRIPTION
**Pull request by qubist**
From db61009

### CODE REVIEW

File: slowmovie.py [hunk 1]
```diff
@@ -105,15 +105,44 @@ def video_info(file):
...
+            "frame_time": frameTime}
 
         videoInfos[file] = info
     return info
 
 
+# Returns the next video in the videos directory, or the first one if there's no current video
+def get_next_video(viddir, currentVideo=None):
+    # Only consider videos in the directory
+    videos = list(filter(supported_filetype, os.listdir(viddir)))
```
Comment (missionfloyd) [inline slowmovie.py:120]:
```suggestion
    videos = sorted(list(filter(supported_filetype, os.listdir(viddir))))
```
Should put this back

Comment (qubist) [reply slowmovie.py]: Thanks ✅

File: slowmovie.py [hunk 2]
```diff
@@ -105,15 +105,44 @@ def video_info(file):
...
+
+    if currentVideo:
+        try:
+            nextIndex = videos.index(currentVideo) + 1
+            # If we're not wrapping around
+            if not nextIndex >= len(videos):
+                return os.path.join(viddir, videos[nextIndex])
+        except ValueError:
+            # Maybe currentVideo is no longer in the list and that's fine
+            pass
```
Comment (missionfloyd) [inline slowmovie.py:134]: How would it end up here? If the current video is deleted/moved/renamed while it's running, wouldn't `generate_frame()` fail?

Comment (robweber) [reply slowmovie.py:134]: I wonder if this was more likely in previous versions when the video directory was read at program start and then never refreshed? The chance of the video being deleted is not 0 but it's extremely small. You'd have to delete it between the generate frame call but prior to this, that window is probably measured in milliseconds.

Comment (qubist) [reply slowmovie.py:134]: Yeah, I think that's what happened here. I'll update the code. Thanks for catching this :)

**Comment by qubist** [comment]
Ready for merge I believe, thoughts?

**Review by robweber — COMMENTED** [review]
I like it, simplifies a lot of that code.

**Review by robweber — APPROVED** [review]

**Pull request merged**